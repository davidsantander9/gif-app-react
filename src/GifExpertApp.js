import React, { useState } from 'react'

const GifExpertApp = () => {

    // const categories = ['One Punch', 'Samurai X', 'Dragon ball'];
    const [categories , setCatergories ] = useState(['One Punch', 'Samurai X', 'Dragon ball']);

    const handleAdd = () => {
        // setCatergories( cats => [...cats, 'Naruto']);
        setCatergories([...categories, 'Naruto'])
    }

    return (
        <>
            <h2>GifExpertApp</h2>
            <hr/>
            <button onClick={ handleAdd }>Agregar</button>
            
            <ol>
                {
                    categories.map( category =>  { return <li key={ category }>{ category }</li> } )
                }
            </ol>
        </>
    )
}

export default GifExpertApp
