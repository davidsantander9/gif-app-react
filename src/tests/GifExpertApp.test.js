import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('<GifExpertApp/>  test', () => {

    test('should correctly display the component', () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect( wrapper ).toMatchSnapshot();
    });    

    test('should display a list of categories', () => {
        const categories = ['Batman', "Halo"];
        const wrapper = shallow(<GifExpertApp defaultCategories = { categories } />);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    });
});