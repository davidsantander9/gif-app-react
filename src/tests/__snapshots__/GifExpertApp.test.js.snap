// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifExpertApp/>  test should correctly display the component 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`<GifExpertApp/>  test should display a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Batman"
      key="Batman"
    />
    <GifGrid
      category="Halo"
      key="Halo"
    />
  </ol>
</Fragment>
`;
